lib/newrelic/telemetry_sdk/span.rb

```diff
@@ -0,0 +1,65 @@
+# encoding: utf-8
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-telemetry-sdk-ruby/blob/main/LICENSE for complete details.
+
+require 'newrelic/telemetry_sdk/util'
+require 'json'
+
+module Newrelic
+  module TelemetrySdk
+    class Span
+      attr_reader :name,
+                  :id,
+                  :trace_id,
+                  :timestamp_ms,
+                  :duration_ms,
+                  :parent_id,
+                  :service_name,
+                  :custom_attributes
+
+      def initialize name,
+                     id: Util.generate_guid(8),
+                     trace_id: Util.generate_guid(16),
+                     timestamp_ms: Util.time_to_ms,
```

Comment:
Just curious, but is the user ever going to want to supply a `timestamp_ms`?  i am thinking, "no" 

And while thinking about this, I realized there's a possible blindspot and potential for bug/inconsistent duration calculations.  Consider if the user is using a Time or DateTime object and passing that value converted to milliseconds.  Then we do our duration calculation based on Process.get_clocktime. How will we handle this?  is there a technically correct way to mix time sources in a duration calc?

Also, perhaps we should rename "timestamp_ms" to "start_time_ms" so it's clear to the user we're asking them to supply the starting_time for the span.

Reply:
To your first question, according to the [Telemetry SDK specs](https://github.com/newrelic/newrelic-telemetry-sdk-specs/blob/master/capabilities.md#spans), users of the SDK must be able to supply a timestamp/start time. My assumption is that this will be commonly used when transforming spans that already exist, e.g. in an exporter.

Your second point is definitely a good one. Even in contexts where newer versions of Ruby are required and monotonic time _could_ be used, `Time` objects are often used instead (including OpenTelemetry). The situation you describe definitely seems possible. I would like to keep the calculation of duration as simple as possible though.

I totally agree with the naming suggestion and will make that change.

Reply:
Discussing "in real life," we decided to make time source configurable, with a Time object being the default since that would seem to be most intuitive.

Reply:
I have added the need for a time source configuration option to https://github.com/newrelic/newrelic-telemetry-sdk-ruby/issues/10.